### PR TITLE
Implement Observation.copy() and tests

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request implements an `Observations.copy()` method and corresponding tests. This addresses part of #4024.

### CODE REVIEW

File: gammapy/data/observations.py
Comment (AtreyeeS) [inline gammapy/data/observations.py:513]: Maybe add "set True" to overwrite parameters?

**Review by AtreyeeS — APPROVED** [review]
Thanks @adonath !

File: gammapy/data/observations.py
Comment (adonath) [reply gammapy/data/observations.py]: Thanks, done!

**Comment by adonath** [comment]
For efficiency I will merge this PR now and not wait for the 2nd review. @AtreyeeS Once this PR is merged please rebase #4024.

**Pull request merged**
